### PR TITLE
feat: split schematics component localization

### PR DESCRIPTION
**Pull request by matthieu-crouzet**
## Proposed change

Split schematics localization of component

### CODE REVIEW

File: packages/@o3r/core/schematics/component/common/localization.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  componentPath: string,
+  options: Pick<NgGenerateComponentSchematicsSchema, 'useLocalization' | 'skipLinter' | 'activateDummy'>,
+  context: SchematicContext
+): Promise<Rule[]> => {
+  let useLocalization = options.useLocalization;
+  let alwaysUseLocalization;
+  try {
+    require.resolve(`${localizationPackageName}/package.json`);
+    if (useLocalization === null && context.interactive) {
+      useLocalization = await askConfirmation('Generate component with Otter localization ?', true);
```
Comment (vscaiceanu-1a) [inline packages/@o3r/core/schematics/component/common/localization.ts:19]:
```suggestion
      useLocalization = await askConfirmation('Generate component with Otter localization?', true);
```

File: packages/@o3r/localization/schematics/localization-to-component/schema.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  "$schema": "http://json-schema.org/schema",
+  "$id": "NgAddLocalizationSchematicsSchema",
+  "title": "Add Otter Localization to an existing component",
+  "description": "Add Otter Localization to an existing component",
+  "properties": {
+    "path": {
+      "type": "string",
+      "description": "Path to the component"
+    },
+    "specFilePath": {
```
Comment (sdo-1A) [inline packages/@o3r/localization/schematics/localization-to-component/schema.json:11]: what is the use of this property?

File: packages/@o3r/localization/schematics/localization-to-component/schema.json [hunk 3]
```diff
@@ -0,0 +1,31 @@
+{
+  "$schema": "http://json-schema.org/schema",
```
Comment (vscaiceanu-1a) [inline packages/@o3r/localization/schematics/localization-to-component/schema.json:2]: should be added to the docs

File: packages/@o3r/localization/schematics/localization-to-component/schema.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import type { JsonObject } from '@angular-devkit/core';
+
+export interface NgAddLocalizationSchematicsSchema extends JsonObject {
+  /** Path to the component */
+  path: string;
+
+  /** Path to spec file of the component */
+  specFilePath: string;
```
Comment (vscaiceanu-1a) [inline packages/@o3r/localization/schematics/localization-to-component/schema.ts:8]: is this used?

File: packages/@o3r/localization/schematics/localization-to-component/schema.json [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+      "description": "Path to the component"
+    },
+    "specFilePath": {
+      "type": "string",
+      "description": "Path to spec file of the component"
+    },
+    "skipLinter": {
+      "type": "boolean",
+      "description": "Skip the linter process",
+      "default": false
```
[the remark this answers is not in the document]
Comment (matthieu-crouzet) [reply packages/@o3r/localization/schematics/localization-to-component/schema.json:18]: yes like all others

File: packages/@o3r/localization/schematics/localization-to-component/schema.json (hunk 2 above)
Comment (matthieu-crouzet) [reply packages/@o3r/localization/schematics/localization-to-component/schema.json:11]: https://github.com/AmadeusITGroup/otter/pull/464/files#diff-c7ba2237ccab109059dc14110a7840ecc6761ba6ec20147eeaa9a421768de162R296

File: packages/@o3r/localization/schematics/localization-to-component/schema.ts (hunk 4 above)
Comment (matthieu-crouzet) [reply packages/@o3r/localization/schematics/localization-to-component/schema.ts:8]: https://github.com/AmadeusITGroup/otter/pull/464/files#diff-c7ba2237ccab109059dc14110a7840ecc6761ba6ec20147eeaa9a421768de162R296

File: packages/@o3r/core/schematics/component/common/localization.ts [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+          }
+        ], 0, null);
+      } else {
+        context.logger.info(`
+          You can add localization later to this component via this command:
+          ng g ${localizationPackageName}:${addLocalizationSchematicName} --path="${componentPath}"
+        `);
+      }
+    }
+  } catch {
```
[the remark this answers is not in the document]
Comment (matthieu-crouzet) [reply packages/@o3r/core/schematics/component/common/localization.ts:45]: Nothing else should throw an error in this try catch

File: packages/@ama-sdk/generator-sdk/src/generators/shell/templates/base/package.json [hunk 7]
```diff
@@ -84,7 +84,7 @@
     "@o3r/eslint-plugin": "^<%= sdkCoreVersion %>",
     "@swc/cli": "^0.1.57",
     "@swc/core": "^1.2.249",
-    "@types/jasmine": "4.3.4",
+    "@types/jasmine": "4.3.5",
```
Comment (kpanot) [inline packages/@ama-sdk/generator-sdk/src/generators/shell/templates/base/package.json:87]: could be `~`

File: packages/@o3r/localization/collection.json [hunk 8]
```diff
@@ -6,6 +6,12 @@
       "factory": "./schematics/ng-add/index#ngAdd",
       "schema": "./schematics/ng-add/schema.json",
       "aliases": ["install", "i"]
+    },
+    "localization-to-component": {
```
Comment (kpanot) [inline packages/@o3r/localization/collection.json:10]: Should be added to readme?

File: packages/@o3r/localization/schematics/localization-to-component/schema.ts (hunk 4 above)
Comment (kpanot) [inline packages/@o3r/localization/schematics/localization-to-component/schema.ts:8]: Is the default value `null` or `empty string`?
Maybe better to explicit it

**Review by kpanot — DISMISSED** [review]

**Review by vscaiceanu-1a — APPROVED** [review]

**Review by kpanot — APPROVED** [review]

**Pull request merged**